apps/frontend/src/pages/locations/[id].tsx

```diff
@@ -84,9 +99,83 @@ const LocationTapModal: React.FC<LocationTapModalProps> = ({
 
 export default function LocationPage() {
   const router = useRouter();
-  const { location } = router.query;
+  const { id } = router.query;
+  const [location, setLocation] = useState<Location | null>(null);
+  const [tapInfo, setTapInfo] = useState<{
+    tapParams: TapParams;
+    tapResponse: ChipTapResponse;
+  } | null>(null);
+  const [tapState, setTapState] = useState(TapState.SUCCESS);
   const [seeFullLeaderboard, setSeeFullLeaderboard] = useState(false);
   const [showTapModal, setShowTapModal] = useState(true);
+  const [weeklyTapDays, setWeeklyTapDays] = useState<number[]>([]);
+
+  useEffect(() => {
+    const fetchLocationAndTapInfo = async () => {
+      const user = await storage.getUser();
+      const session = await storage.getSession();
+      const location = user?.locations?.[id as string];
+      if (!user || !session || !location) {
+        console.error("Location not found");
+        toast.error("Location not found");
+        router.push("/profile");
+        return;
+      }
+      setLocation(location);
+
+      // Compute the days of the week that the user has tapped in
+      const tapDays = computeWeeklyTapDays(
+        location.taps?.map((tap) => new Date(tap.timestamp))
+      );
+      setWeeklyTapDays(tapDays);
+
+      const savedTapInfo = await storage.loadSavedTapInfo();
+      // Delete saved tap info after fetching
+      await storage.deleteSavedTapInfo();
```

Comment:
Delete within the if case? Should we still delete the tap info if it's not equal to the id?

Reply:
yes i think we should, it shouldnt happen but i think case it does i think it makes sense to delete it